lib/ssl/src/ssl_connection.hrl

```diff
@@ -31,11 +31,13 @@
 -include("ssl_handshake.hrl").
 -include("ssl_srp.hrl").
 -include("ssl_cipher.hrl").
+-include("ssl_api.hrl").
 -include("ssl_alert.hrl").
 -include_lib("public_key/include/public_key.hrl").
 
 -record(static_env, {
                      role                  :: client | server,
+                     user_socket           :: #sslsocket{},
```

Comment:
It is the socket of the user process as opposed to the internal socket owned by the TLS (SSL)  socket owner process that is the receiver of data. But I guess we could call it ssl_socket if you think that is clearer.

Reply:
No I like user_socket even more now.

Reply:
user_socket, might also mean TCP socket provided by user in upgrade scenario.

but maybe it is fine. I discussed it with Dan and he convinced me to large extent.

Reply:
No it will not mean TCP socket provided in upgrade scenario, that will be the internal socket once the upgrade is complete.  I think I prefer user socket too as it is the socket provided by the ssl application API for the user process.